### PR TITLE
Change email restriction error message

### LINKED ISSUE #11760: Change email restriction error message
Current message:
`The email address you used for your developer account is not allowed for add-on submission.`

@eviljeff's proposal:
`The email address used for your account is not allowed for add-on submission.`

### PR DESCRIPTION
**Pull request by megha-kumarii**
Fixes #11760 

cc @championshuttler

### CODE REVIEW

**Comment by megha-kumarii** [comment]
@diox can you please help me with that?

**Comment by diox** [comment]
Our unit tests run through travis-ci (you can ignore the failing *integration* tests). Click on the details link, and look at the jobs that are failing. It will let you know which tests are failing and which files contain those failures if you look at the log carefully.

**Comment by megha-kumarii** [comment]
@diox I looked into it and got which tests are failing but did not get the reason for failing

**Comment by megha-kumarii** [comment]
@diox I looked into it and got which tests are failing but did not get the reason for failing

File: src/olympia/users/models.py [hunk 1]
```diff
@@ -711,7 +711,7 @@ class EmailUserRestriction(NormalizeEmailMixin, ModelBase):
             ' Please note that we do not include "@" in the match so you '
             ' should do that in the pattern.'))
 
-    error_message = _('The email address you used for your developer account'
+    error_message = _('The email address used for your account is not '
                       ' is not allowed for add-on submission.')
```
Comment (championshuttler) [inline src/olympia/users/models.py]: Here you also need to make some changes

Comment (megha-kumarii) [reply src/olympia/users/models.py]: Thanks

**Comment by caitmuenster** [comment]
Woohoo congrats on landing your first PR, @megha-kumarii! 🎉 Your contribution has been added to our [recognition wiki](https://wiki.mozilla.org/Add-ons/Contribute/Recognition) and your Mozillians profile has been vouched. 😎

Welcome onboard! I look forward to seeing you around the project.

**Pull request merged**